### ISSUE TITLE
Please Add hostNetwork to deployment

### ISSUE THREAD

**Issue by esteban1983cl** [banzaicloud/banzai-charts#1060]
**Is your feature request related to a problem? Please describe.**
We are working with AWS EKS and we had few ip addreses in AWS Subnets to support pods. 
So we replace AWS CNI plugin by Weave Net CNI Plugin.

The issue is when you deploy kube-metrics-adapter it creates a apiservice that doesnt work when you don't use default CNI plugin. Because k8s control plane don't see this apiservice.

Example:
```
kubectl get apiservice
NAME                                   SERVICE                           AVAILABLE                      AGE
v1.                                    Local                             True                           349d
v1.apps                                Local                             True                           349d
v1.authentication.k8s.io               Local                             True                           349d
v1.authorization.k8s.io                Local                             True                           349d
v1.autoscaling                         Local                             True                           349d
v1.batch                               Local                             True                           349d
v1.monitoring.coreos.com               Local                             True                           40d
v1.networking.k8s.io                   Local                             True                           349d
v1.rbac.authorization.k8s.io           Local                             True                           349d
v1.storage.k8s.io                      Local                             True                           349d
v1.velero.io                           Local                             True                           40d
v1alpha1.crd.k8s.amazonaws.com         Local                             True                           40d
v1alpha1.wordpress.presslabs.org       Local                             True                           20d
v1beta1.admissionregistration.k8s.io   Local                             True                           349d
v1beta1.apiextensions.k8s.io           Local                             True                           349d
v1beta1.apps                           Local                             True                           349d
v1beta1.authentication.k8s.io          Local                             True                           349d
v1beta1.authorization.k8s.io           Local                             True                           349d
v1beta1.batch                          Local                             True                           349d
v1beta1.certificates.k8s.io            Local                             True                           349d
v1beta1.coordination.k8s.io            Local                             True                           331d
v1beta1.events.k8s.io                  Local                             True                           349d
v1beta1.extensions                     Local                             True                           349d
v1beta1.external.metrics.k8s.io        monitoring/kube-metrics-adapter   False (FailedDiscoveryCheck)   21h
v1beta1.metrics.k8s.io                 kube-system/metrics-server        True                           22d
v1beta1.policy                         Local                             True                           349d
v1beta1.rbac.authorization.k8s.io      Local                             True                           349d
v1beta1.scheduling.k8s.io              Local                             True                           349d
v1beta1.storage.k8s.io                 Local                             True                           349d
v1beta2.apps                           Local                             True                           349d
v2beta1.autoscaling                    Local                             True                           349d
v2beta2.autoscaling                    Local                             True                           331d
```

I explain this because we had the same issue with metrics-server

**Describe the solution you'd like to see**
To solve this, you need add hostNetwork: true in your deployment parameter for apiserver "see" the service

**Describe alternatives you've considered**
N/A

**Additional context**
If it's helpful https://medium.com/codeops/installing-weave-cni-on-aws-eks-51c2e6b7abc8

**Comment by esteban1983cl** [comment]
Evidence:

When you add hostNetwork: true to deployment and re-create apiservice resources, you can see that apiservices are registered successfully

```
NAME                                   SERVICE                           AVAILABLE   AGE
v1.                                    Local                             True        349d
v1.apps                                Local                             True        349d
v1.authentication.k8s.io               Local                             True        349d
v1.authorization.k8s.io                Local                             True        349d
v1.autoscaling                         Local                             True        349d
v1.batch                               Local                             True        349d
v1.monitoring.coreos.com               Local                             True        41d
v1.networking.k8s.io                   Local                             True        349d
v1.rbac.authorization.k8s.io           Local                             True        349d
v1.storage.k8s.io                      Local                             True        349d
v1.velero.io                           Local                             True        41d
v1alpha1.crd.k8s.amazonaws.com         Local                             True        41d
v1alpha1.wordpress.presslabs.org       Local                             True        20d
v1beta1.admissionregistration.k8s.io   Local                             True        349d
v1beta1.apiextensions.k8s.io           Local                             True        349d
v1beta1.apps                           Local                             True        349d
v1beta1.authentication.k8s.io          Local                             True        349d
v1beta1.authorization.k8s.io           Local                             True        349d
v1beta1.batch                          Local                             True        349d
v1beta1.certificates.k8s.io            Local                             True        349d
v1beta1.coordination.k8s.io            Local                             True        331d
v1beta1.custom.metrics.k8s.io          monitoring/kube-metrics-adapter   True        6s
v1beta1.events.k8s.io                  Local                             True        349d
v1beta1.extensions                     Local                             True        349d
v1beta1.external.metrics.k8s.io        monitoring/kube-metrics-adapter   True        5s
v1beta1.metrics.k8s.io                 kube-system/metrics-server        True        22d
v1beta1.policy                         Local                             True        349d
v1beta1.rbac.authorization.k8s.io      Local                             True        349d
v1beta1.scheduling.k8s.io              Local                             True        349d
v1beta1.storage.k8s.io                 Local                             True        349d
v1beta2.apps                           Local                             True        349d
v2beta1.autoscaling                    Local                             True        349d
v2beta2.autoscaling                    Local                             True        331d
```

**Comment by sancyx** [comment]
EnableHostNetwork param is available in version 0.1.3.
